File: leetcode/191. Number of 1 Bits/OneBits.go
// Solution one:

func hammingWeight(num uint32) int {
	var sum uint32
	for num > 0 {
		var temp uint32
		temp = num % 2
		sum += temp
		num = num >> 1
	}
	return int(sum)
}

//********************************

//Solution two:

func hammingWeight(num uint32) int {
	var sum uint32
	for num > 0 {
		num = num & (num - 1)
		sum++
	}
	return int(sum)
}

File: leetcode/191. Number of 1 Bits/OneBits.cpp
class Solution
{
public:
    int hammingWeight(uint32_t n)
    {
        uint32_t sum = 0;
        while (n > 0)
        {
            uint32_t temp = 0;
            temp = n % 2;
            sum += temp;
            n >>= 1;
        }
        return int(sum);
    }
};